pkgs/applications/window-managers/cosmic/panel/default.nix

```
@@ -0,0 +1,51 @@
+{ lib, stdenv, fetchFromGitHub, cargo, just, pkg-config, rustPlatform
+, libglvnd, libxkbcommon, wayland
+, unstableGitUpdater
+}:
+
+stdenv.mkDerivation {
+  pname = "cosmic-panel";
+  version = "unstable-2023-08-03";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = "cosmic-panel";
+    rev = "edfd24ed3b712de397057906924e4f7e8b6252c4";
+    hash = "sha256-H3QuiP7Og69wm9yCX/uoSG0aQ3B/61q9Sdj+rW4KZMU=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cosmic-config-0.1.0" = "sha256-ENdQkLo4MSH/oe6vFr8qByWehch0nMngBr9/6hF7iVk=";
+      "cosmic-notifications-util-0.1.0" = "sha256-KnPQdrMpzA05v4bt0Fz9fbcKdC0cSU60Hv7wqrthIaw=";
+      "directories-4.0.1" = "sha256-4M8WstNq5I7UduIUZI9q1R9oazp7MDBRBRBHZv6iGWI=";
+      "launch-pad-0.1.0" = "sha256-gFtUtrD/cUVpLxPvg6iLxxAK97LTlvI4uLxo06UYIU4=";
+      "smithay-0.3.0" = "sha256-qjnpDRzRL0FZ+QYjtD5JeITIY+sfY/yqETNtVT88Ge0=";
+      "smithay-client-toolkit-0.17.0" = "sha256-v3FxzDypxSfbEU50+oDoqrGWPm+S+kDZQq//3Q4DDRU=";
+      "xdg-shell-wrapper-0.1.0" = "sha256-Otxp8D5dNZl70K1ZIBswGj6K5soGVbVim7gutUHkBvw=";
+    };
+  };
+
+  nativeBuildInputs = [ cargo just pkg-config rustPlatform.cargoSetupHook ];
```

```suggestion
  nativeBuildInputs = [ cargo just pkg-config ];
```

Unfortunately `foo.bar` references (still) don't splice correctly, so you can't use `rustPlatform.cargoSetupHook` in `nativeBuildInputs`.  But you should get the hook automatically from `cargo`.